### PR TITLE
Move S.N.Vector APIs to corelib and expose them from System.Runtime

### PR DESCRIPTION
**Pull request by ahsonkhan**
Depends on https://github.com/dotnet/coreclr/pull/15821 going through first.

Setting up the ground work for: https://github.com/dotnet/corefx/issues/25182#issuecomment-343614677

cc @jkotas, @KrzysztofCwalina, @safern, @ViktorHofer, @eerhardt

### CODE REVIEW

File: src/System.Numerics.Vectors/ref/System.Numerics.Vectors.csproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
     <ProjectGuid>{650277B5-9423-4ACE-BB54-2659995B21C7}</ProjectGuid>
-    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'netfx' OR '$(TargetGroup)' == 'net46'">true</IsPartialFacadeAssembly>
+    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'netfx' Or '$(TargetGroup)' == 'net46' Or '$(TargetGroup)' == 'netcoreapp' Or '$(TargetGroup)' == 'uap'">true</IsPartialFacadeAssembly>
```
Comment (safern) [inline src/System.Numerics.Vectors/ref/System.Numerics.Vectors.csproj]: I don't think this is a proper usage of IsPartialFacadeAssembly I think for targetgroup != netfx and != net46 you would need to add the typeforwards to System.Runtime manually. That is why the build is failing from AFAIK but not sure,  @weshaggard knows more about the partial facade generation.

**Comment by weshaggard** [comment]
I applied the "No Merge" label to this PR as I want to make an explicit decision about moving Vectors into corelib before we move forward and merge this. The discussion is happening at https://github.com/dotnet/corefx/issues/25182.

File: src/System.Numerics.Vectors/ref/System.Numerics.Vectors.csproj (hunk 1 above)
Comment (weshaggard) [reply src/System.Numerics.Vectors/ref/System.Numerics.Vectors.csproj]: If we do go forward with this you don't want to turn the contract into a partial facade unless you are going to move the API's to System.Runtime which I don't think we are planning to do.

File: src/System.Numerics.Vectors/src/System.Numerics.Vectors.csproj [hunk 2]
```diff
@@ -45,29 +47,21 @@
...
-    <Compile Include="System\Numerics\Quaternion.cs" />
-    <Compile Include="System\Numerics\Vector2.cs" />
-    <Compile Include="System\Numerics\Vector2_Intrinsics.cs" />
-    <Compile Include="System\Numerics\Vector3.cs" />
-    <Compile Include="System\Numerics\Vector3_Intrinsics.cs" />
-    <Compile Include="System\Numerics\Vector4.cs" />
-    <Compile Include="System\Numerics\Vector4_Intrinsics.cs" />
+  <ItemGroup Condition="'$(TargetGroup)' == 'netcoreapp' Or '$(TargetGroup)' == 'uap'">
+    <ReferenceFromRuntime Include="System.Private.CoreLib" />
+    <ProjectReference Include="..\..\System.Runtime\src\System.Runtime.csproj" />
```
Comment (weshaggard) [inline src/System.Numerics.Vectors/src/System.Numerics.Vectors.csproj]: Why do you need to reference System.Runtime here?

**Comment by karelz** [comment]
@ahsonkhan any plans to push the PR forward?

**Comment by eerhardt** [comment]
@karelz - I'm going to take this work forward.  Let's close this PR for now, and if it makes sense I'll re-open it when the coreclr work is completed.

**Pull request closed without merging**